src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -159,12 +158,17 @@ public virtual async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsyn
 
             CounterfactualLoggers.TransitiveDependencies.EmitIfNeeded(); // Emit only one event per VS session
             IEnumerable<TransitivePackageReference> transitivePackagesWithOrigins;
-            if (await ExperimentUtility.IsTransitiveOriginExpEnabled.GetValueAsync(token))
+            if (await ExperimentUtility.IsTransitiveOriginExpEnabled.GetValueAsync(token) && useTransitiveOrigins)
             {
                 // Compute Transitive Origins
                 Dictionary<string, TransitiveEntry> transitiveOrigins;
                 if (IsInstalledAndTransitiveComputationNeeded || TransitiveOriginsCache == null)
                 {
+                    if (targetsList == null) // special case
+                    {
+                        targetsList = (await GetTargetsListAsync(assetsPath, token))?.ToList();
+                    }
```

Comment:
I see that previously we get `targetsList` the same way as here, can you explain me what is changing that will make `GetTargetsListAsync` return a different result from the above one?

Reply:
See https://github.com/NuGet/NuGet.Client/pull/4788/files#r972353643